### PR TITLE
Restructure frontend

### PR DESCRIPTION
**Pull request by mathijshenquet**
Highlights:
- Uses typescript for type checking now
- Uses webpack to put all the javascript in a bundle
- The website (almost) works on IE11 now

Further comments:
- Docker compose also starts a webpack process now. Development should still work with `docker-compose up` add `--build` if you added packages to yarn
- The bundle should be correctly built and pushed to staging/production by github actions (tested with staging)

### CODE REVIEW

**Comment by mathijshenquet** [comment]
It's currently running on http://staging.epidemicforecasting.org/.

**Comment by hnykda** [comment]
I generally like it if we talk about tech stuff. 

Does it make sense to still serve files from Fastapi server? What about just using something like https://github.com/mozilla/nunjucks and express.js to simplify the stack to JS only?

**Comment by mathijshenquet** [comment]
Sorry, ill try to be more communicative. I think we could move to a full JS stack in the long run. However it is not very unnatural to keep the javascript compiling and the webserver orthogonal.

**Comment by hnykda** [comment]
> However it is not very unnatural to keep the javascript compiling and the webserver orthogonal
I don't think there is someone who would suffer from this. Given how little logic there is on the backend, I think we can move it to the same stack if you and Jon think it's a good move too.

**Comment by mathijshenquet** [comment]
I was least sure about how to do the separate pages. Currently everything is in one bundle which I think is the right approach since almost all js is for our landing page and this way everything is cached afterwards. It now executes code depending on if it can find some elements which should probably be done differently. Maybe webpack can expose some global functions

**Comment by wolverdude** [comment]
Re: moving to full TS stack, I'm all for that as well. I can help with the refactor when we get there.

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
     - name: Prettify the JS Code
       uses: creyD/prettier_action@v2.0
       with:
-        prettier_options: 'prettier --check "{src/server/static/**, src/server/templates/**}"'
+        prettier_options: 'prettier --check "{src/scripts/**, src/server/**, src/server/templates/**}"'
```
[the remark this answers is not in the document]
Comment (mathijshenquet) [reply .github/workflows/pythonapp.yml:30]: I went with `frontend`

**Comment by hnykda** [comment]
OK, please add what needs to be done for production (I guess at least the gsutil upload in CI), rebase, and happy to get it merged?

**Comment by mathijshenquet** [comment]
@hnykda this is already done I think.

**Comment by hnykda** [comment]
Ah, you are right, sorry, I overlooked that. All good

**Pull request merged**